apigw-websocket-api-lambda-authorizer/template.yaml

```diff
@@ -0,0 +1,272 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: An AWS Serverless Application to demonstrates Cognito authentication with Amazon API Gateway WebSocket API.
+Parameters:
+  ConnectionMappingTableName:
+    Type: String
+    Default: SimpleChatConnections
+    Description: The name of the new DynamoDB to store connection identifiers for
+      each connected clients. Minimum 3 characters.
+    MinLength: '3'
+    MaxLength: '50'
+    AllowedPattern: ^[A-Za-z_]+$
+Resources:
+
+ # API Gateway Websocket API, Deployment & Stage
+  SimpleChatWebSocketApi:
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Name: SimpleChatWebSocket
+      ProtocolType: WEBSOCKET
+      RouteSelectionExpression: $request.body.message
+  Deployment:
+    Type: AWS::ApiGatewayV2::Deployment
+    DependsOn:
+      - ConnectRoute
+      - SendMessageRoute
+      - DisconnectRoute
+    Properties:
+      ApiId: !Ref 'SimpleChatWebSocketApi'
+  Stage:
+    Type: AWS::ApiGatewayV2::Stage
+    Properties:
+      ApiId: !Ref 'SimpleChatWebSocketApi'
+      DeploymentId: !Ref 'Deployment'
+      StageName: Prod
+
+ # Lambda functions for OnConnect, OnDisconnect, SendMessage
+  OnConnectFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: WebSocketAPI::WebSocketAPI.Functions::OnConnectHandler
+      Runtime: dotnet6
+      CodeUri: './src/WebSocketAPI'
+      MemorySize: 256
+      Timeout: 30
+      Policies:
+        - DynamoDBCrudPolicy:
+            TableName: !Ref 'ConnectionMappingTableName'
+      Environment:
+        Variables:
+          TABLE_NAME: !Ref 'ConnectionMappingTableName'
+  OnDisconnectFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: WebSocketAPI::WebSocketAPI.Functions::OnDisconnectHandler
+      Runtime: dotnet6
+      CodeUri: './src/WebSocketAPI'
+      MemorySize: 256
+      Timeout: 30
+      Policies:
+        - DynamoDBCrudPolicy:
+            TableName: !Ref 'ConnectionMappingTableName'
+      Environment:
+        Variables:
+          TABLE_NAME: !Ref 'ConnectionMappingTableName'
+  SendMessageFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: WebSocketAPI::WebSocketAPI.Functions::SendMessageHandler
+      Runtime: dotnet6
+      CodeUri: './src/WebSocketAPI'
+      MemorySize: 256
+      Timeout: 30
+      Policies:
+        - DynamoDBCrudPolicy:
+            TableName: !Ref 'ConnectionMappingTableName'
+        - Statement:
+            Effect: Allow
+            Action: execute-api:ManageConnections
+            Resource: arn:aws:execute-api:*:*:*/@connections/*
```

Comment:
Close scope down this policy a bit, use pseudo parameters for the accountId and region: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html

Reply:
Done